exchanges/stream/websocket_connection.go

```diff
@@ -123,8 +128,11 @@ func (w *WebsocketConnection) SendRawMessage(messageType int, message []byte) er
 	if w.Verbose {
 		log.Debugf(log.WebsocketMgr, "%v websocket connection: sending message [%s]\n", w.ExchangeName, message)
 	}
-	if w.RateLimit > 0 {
-		time.Sleep(time.Duration(w.RateLimit) * time.Millisecond)
+	if w.RateLimit != nil {
+		err := request.RateLimit(ctx, w.RateLimit)
+		if err != nil {
+			return fmt.Errorf("%s websocket connection: rate limit error: %w", w.ExchangeName, err)
+		}
 		if !w.IsConnected() {
```

Comment:
Not you, but this seems a little redundant and could be removed due to checking the connection twice which seems a little redundant

Reply:
This needs to stay because between the time we limit the request and then wakeup the routine, the connection could be closed. I will add a note.

Reply:
Wait no its not. 😆